### PR TITLE
DBZ-1203 Alter Enum Column When Changing Character Set

### PR DESCRIPTION
**Pull request by Naros**
https://issues.jboss.org/browse/DBZ-1203

### CODE REVIEW

**Comment by Naros** [comment]
Upstream issue https://github.com/antlr/grammars-v4/pull/1384

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java [hunk 1]
```diff
@@ -568,6 +568,28 @@ public void parseGeomCollection() {
...
+                        "    PRIMARY KEY (`id`)\n" +
+                        ");\n" +
+                        "\n" +
+                        "ALTER TABLE `test_stations_10`\n" +
+                        "    MODIFY COLUMN `type` ENUM('station', 'post_office', 'plane', 'ahihi_dongok', 'now')\n" +
+                        "    CHARACTER SET 'utf8' COLLATE 'utf8_unicode_ci' NOT NULL DEFAULT 'station';\n";
+
+        parser.parse(ddl, tables);
+        assertThat(tables.size()).isEqualTo(1);
+        assertThat(tables.forTable(new TableId(null, null, "test_stations_10"))).isNotNull();
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java]: Perhaps add an assertion of the actual enum constants (if doable, I'm not sure how it is exposed, might be the "type expression" of the column)?

Comment (jpechane) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java]: @gunnarmorling  It is tested in `MySqlEnumColumnIT.java` - at the schema level

**Comment by jpechane** [comment]
@Naros Thanks a lot! @gunnarmorling It is definitely good to be merged.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java]: Ah yes, thanks.

**Comment by gunnarmorling** [comment]
Great work; thanks a lot, @Naros!

**Pull request merged**